### DISCUSSION TITLE
Victron 75/15 keine Liveansicht

### DISCUSSION THREAD

**Discussion by larsfcfa** [hoylabs/OpenDTU-OnBattery#653] [Q&A] [answered]
Hallo Leute,

ich habe heute nach erfolgreichem openDTU-Test auch das onBattery probiert.
Die Kommunikation mit dem Hoymiles über cmt2300a klappt bei beiden Varianten.
Nachdem ich onBattery installiert habe, sehe ich kein Eintrag dazu im Live-View, keinerlei Traffic von ve.direct in der Konsole.
Wo sollte ich denn sehen, dass der Victron korrekt eingebunden ist?

-ich habe TX (weiß) und GND über einen Level-Shifter mit dem ESP32 (an RX) verbunden
-in der config.json entsprechend GPIO22 eingetragen
-ich habe aus dem Git das fertige opendtu-generic.bin auf den ESP übertragen
-ich habe über das interne Firmeware-Update fertige opendtu-generic.bin auf den ESP übertragen
-ich habe über platformio ein aus dem https://github.com/helgeerbe/OpenDTU-OnBattery.git ein git-clone erstellt, kompiliert und als internes Firmeware-Update aufgespielt
-TX steht im MPPT auf normale Kommunikation

Keine der Varianten hat zu einer Lösung geführt.

Hat jemand noch eine Idee?

**Comment by schlimmchen** [comment]
> -ich habe TX (weiß) und GND über einen Level-Shifter mit dem ESP32 (an RX) verbunden

Weiß? Warum ausgerechne weiß? Den Farben ist nicht zu trauen. Hast du einmal geprüft ob diese weiße Ader wirkin am richtigen Pin rauskommt?

> -in der config.json entsprechend GPIO22 eingetragen

Meinst du `pin_mapping.json`? Steht beim Starten des ESP32 in der seriellen Konsole am PC über USB, dass VE.Direct initialisiert wurde, oder steht da "invalid pin mapping"?

Kennst du https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles und https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface und hast die Seiten gelesen und verstanden?

Wie hast du den Level-Shifter angeschlossen? Bist du sicher, dass du da keinen Fehler gemacht hast? Teile ggf. ein Foto.

**Reply by larsfcfa** [reply]
Hallo Bernhard,

vielen Dank für deine Antwort, lass mich versuchen bestmöglich die Punkte zu beantworten:

> -Weiß? Warum ausgerechne weiß? Den Farben ist nicht zu trauen. Hast du einmal geprüft ob diese weiße Ader wirkin am richtigen Pin rauskommt

ich habe keinen Logic Analyzer verwendet, aber für mich scheint es - wenn die Pinbelegung korrekt ist - zu stimmen (siehe Fotos)
Mit welcher Baudrate erfolgt denn die Übertragung?

> Meinst du pin_mapping.json? Steht beim Starten des ESP32 in der seriellen Konsole am PC über USB, dass VE.Direct initialisiert wurde, oder steht da "invalid pin mapping"?

-> hier meine json, in der Info/Konsole von der openDTU-site steht nur was von fetch inverter und die TX-Übertragung bei 865MHz - das sollte ja das cmt2300a Modul sein. Vom Victron sehe ich dort nichts
Nur zum Verständnis: wenn der Victron erkannt wird, erscheint der auch automatisch auf dem Live-View?

[cmt_vectron_config.json](https://github.com/helgeerbe/OpenDTU-OnBattery/files/14223813/cmt_vectron_config.json)

Der Level-Shifter ist eine simple Transistorschaltung aus der Bucht, nichts für MHz-Signale, sollte aber für Serielle Übertragung ausreichen.
-weiß (TX) und schwarz (GND) auf die 5V Seite, orange geht dann als 3.3V RX an den ESP32
Platine ist quick und dirty, ich weiß, sollte aber funktionieren
![level shfter all](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/46619108/fbe890b5-12a9-4ff6-955a-daf9658fda4f)



> Kennst du https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles und https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface und hast die Seiten gelesen und verstanden?

Das habe ich mir durchgelesen und hoffentlich auch verstanden

![Victron_Ve_direct](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/46619108/b5c538d5-4d0f-4850-b24f-ed8f2a4b37a0)
![Victron_Verkabelung](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/46619108/0aacafd8-9807-4282-bd54-94139573fe0c)

**Comment by sentinel1111** [comment, marked as answer]
Unter "Einstellungen" ist VE.Direkt auch aktiviert?

**Reply by larsfcfa** [reply]
Hallo,
vielen Dank.

ja VE.Direct ist in den Einstellungen aktiviert, nach mehrmaligem Neustart ist jetzt auch der Victron in der Live-Ansicht zu sehen.
Allerdings klappt die serielle Übertragung immer noch nicht, ich habe schon andere GPIOs ausprobiert (entsprechend in der PIN.Config angepasst), aber es kommt scheinbar nichts an, auch nicht in der Konsole. 
Hat jemand noch eine Idee?
Mit welcher Baud-Rate läuft denn die Kommunikation, wenn ich mal mit einen Logic Analyzer mitlauschen möchte?

**Reply by larsfcfa** [reply]
19230 Bits/s?

**Reply by larsfcfa** [reply]
Vielen Dank, am Ende klappt es nun!

-zunächst einen schlechten Kontakt am Level Shifter beseitigt
-dann das serielle Signal mit dem Logic Analyzer hinter dem Level Shifter gemessen
-dann noch einen Wackelkontakt am ESP32 beseitigt - die Dupont Kabel sind Schrott...
-komischerweise kann ich aber das Signal mit dem simplen Logic Analyzer nicht mehr messen, wenn das Kabel am ESP32 hängt - warum?

Aber wie gesagt, es klappt nun!

**Comment by larsfcfa** [comment]
Hallo

mit dem Level-Shifter war ein Kontakt nicht ok, separat kann ich nun das Serielle Signal mit dem Logic Analyzer hinter dem Level Shifter sauber messen.
Komischerweise kann ich das Signal nicht mehr messen, wenn ich es mit dem ESP32 verbunden habe.
Nach einem Neustart steht nun aber die Serielle Verbindung zwischen OpenDTU und dem MPPT. Da gab es auch noch einen Wackelkontakt, die Dupont Kabel sind echt Schrott!! (zumindest als Einzelkontakt) 
Vielen Dank, am Ende klappt es nun!

**Comment by larsfcfa** [comment]
Vielen Dank, am Ende klappt es nun!

**Reply by schlimmchen** [reply]
Danke für die Rückmeldung. Und schau dir einen ADUM1201 an, falls du Lust hast dieses Setup auf ein neues Level zu heben. :wink:

**Comment by hoschiking** [comment]
Vorsicht bei Verwendung von seriellen Verbindungen ohne galvanische Trennung.
Falls dein BMS den Minupol trennt ( das tun zumindest alle Chinesen) raucht die alles was an der UART hängt ab.
Deswegen ist das originale Victron Datenkabel galvanisch getrennt. 
Bei DIY empfiehl sich der ADUM1201

**Reply by schlimmchen** [reply]
> Falls dein BMS den Minupol trennt ( das tun zumindest alle Chinesen) raucht die alles was an der UART hängt ab.

Ich fühle mich elektrotechnisch zwar einigermaßen fit, aber das begreife ich noch nicht. Wo entsteht durch die Trennung der Batterie ein ungewolltes Potential?

**Reply by hoschiking** [reply]
Vom BMS zum ESP über die UART Schnittstelle. Auf beiden Seiten ist GND doch mit dem -Pol verbunden.
Das gleiche gilt dann für die Verbindung vom ESP zum MPPT. Der Strom zum Hoymiles und oder MPPT fließt dann genau über diesen Pfad. Es gibt genug Berichte im Netz und ist bei Hardwareentwicklern allgemein bekannt
